devices/uart-stm32/uart.c

```diff
@@ -32,11 +32,57 @@ struct {
 	uart_t uarts[UART_MAX_CNT];
 } uart_common;
 
-
-enum { cr1 = 0, cr2, cr3, brr, gtpr, rtor, rqr, isr, icr, rdr, tdr };
-
-
-static int uartLut[] = { UART1, UART2, UART3, UART4, UART5 };
+/* clang-format off */
+enum { cr1 = 0, cr2, cr3, brr, gtpr, rtor, rqr, isr, icr, rdr, tdr, presc };
+/* clang-format on */
+
+
+#if defined(__CPU_STM32N6)
+/* Values for selecting the peripheral clock for an UART */
+enum {
+	uart_clk_sel_pclk = 0, /* pclk1 or pclk2 depending on peripheral */
+	uart_clk_sel_per_ck,
+	uart_clk_sel_ic9_ck,
+	uart_clk_sel_ic14_ck,
+	uart_clk_sel_lse_ck,
+	uart_clk_sel_msi_ck,
+	uart_clk_sel_hsi_div_ck,
+};
+#endif
+
+
+static int uartLut[UART_MAX_CNT] = {
+#ifdef UART1
+	UART1,
+#endif
+#ifdef UART2
+	UART2,
+#endif
+#ifdef UART3
+	UART3,
+#endif
+#ifdef UART4
+	UART4,
+#endif
+#ifdef UART5
+	UART5,
+#endif
+#ifdef UART6
+	UART6,
+#endif
+#ifdef UART7
+	UART7,
+#endif
+#ifdef UART8
+	UART8,
+#endif
+#ifdef UART9
+	UART9,
+#endif
+#ifdef UART10
+	UART10,
+#endif
```

Comment:
I don't think this is right - if some (e.g. UART5) is not defined, then every next UART is shifted to the lower position.

IMHO we should require all UARTx to be defined